numpy/core/records.py

```diff
@@ -610,8 +612,8 @@ def fromarrays(arrayList, dtype=None, shape=None, formats=None,
         for obj in arrayList:
             if not isinstance(obj, ndarray):
                 raise ValueError("item in the array list must be an ndarray.")
-            formats.append(obj.dtype.str)
-        formats = ','.join(formats)
+            shape = obj.shape[1:] or ''
```

Comment:
is shape meant to be a variable local to this block?

Reply:
You are right the loop variable should not overwrite the _shape_ variable. I've fixed this in the latest commit